raiden/tests/integration/test_echo_node.py

```diff
@@ -39,24 +45,23 @@ def test_event_transfer_received_success(token_addresses, raiden_chain, skip_if_
 
     # sleep is for the receiver's node to have time to process all events
     gevent.sleep(1)
-    events = receiver_app.raiden.wal.storage.get_events_by_block(0, 'latest')
-    events = [e[1] for e in events]
-
-    assert must_contain_entry(
-        events,
-        EventTransferReceivedSuccess,
-        {'amount': 1, 'initiator': app0.raiden.address},
-    )
-    assert must_contain_entry(
-        events,
-        EventTransferReceivedSuccess,
-        {'amount': 2, 'initiator': app1.raiden.address},
-    )
-    assert must_contain_entry(
-        events,
-        EventTransferReceivedSuccess,
-        {'amount': 3, 'initiator': app2.raiden.address},
-    )
+
+    def test_events(amount, address):
+        events = receiver_app.raiden.wal.storage.get_events_by_block(0, 'latest')
+        events = [e[1] for e in events]
+        return must_contain_entry(
+            events,
+            EventTransferReceivedSuccess,
+            {'amount': amount, 'initiator': address},
+        )
+
+    amounts = [1, 2, 3]
+    addrs = [app0.raiden.address, app1.raiden.address, app2.raiden.address]
+    for amount, address in zip(amounts, addrs):
+        wait_until(
```

Comment:
Please, `assert` here, `wait_until` wil return `None` if it can't get a truty value on time, or the value if so.

Reply:
done!